api/v3/Payment.php

```diff
@@ -0,0 +1,291 @@
+ |                                                                    |
+ | CiviCRM is free software; you can copy, modify, and distribute it  |
+ | under the terms of the GNU Affero General Public License           |
+ | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ * This api exposes CiviCRM Contribution Payment records.
+ *
+ * @package CiviCRM_APIv3
+ */
+
+/**
+ * Retrieve a set of financial transactions which are payments.
+ *
+ * @param array $params
+ *  Input parameters.
+ *
+ * @return array
+ *   Array of financial transactions which are payments, if error an array with an error id and error message
+ */
+function civicrm_api3_payment_get($params) {
+  $ft = array();
+  if (CRM_Utils_Array::value('contribution_id', $params)) {
+    $params['entity_id'] = $params['contribution_id'];
+  }
+  $params['entity_table'] = 'civicrm_contribution';
+  $eft = civicrm_api3('EntityFinancialTrxn', 'get', $params);
+  if (!empty($eft['values'])) {
+    foreach ($eft['values'] as $efts) {
+      $eftids[] = $efts['financial_trxn_id'];
+      $map[$efts['financial_trxn_id']] = $efts['entity_id'];
+    }
+    $ftParams = array(
+      'id' => array( 'IN' => $eftids ),
+      'is_payment' => 1,
+    );
+    $ft = civicrm_api3('FinancialTrxn', 'get', $ftParams);
+    foreach ($ft['values'] as &$values) {
+      $values['contribution_id'] = $map[$values['id']];
+    }
+  }
+  return civicrm_api3_create_success($ft['values'], $params, 'Payment', 'cancel');
+}
+
+/**
+ * Delete a payment.
+ *
+ * @param array $params
+ *   Input parameters.
+ *
+ * @throws API_Exception
+ * @return array
+ *   Api result array
+ */
+function civicrm_api3_payment_delete(&$params) {
+  $params['id'] = isset($params['civicrm_financial_trxn_id']) ? $params['civicrm_financial_trxn_id'] : $params['payment_id'];
+  if (!CRM_Core_DAO::getFieldValue('CRM_Financial_DAO_FinancialTrxn', $params['id'], 'id')) {
+    return civicrm_api3_create_error(ts('You need to supply a valid Payment ID to delete a payment'));
```

Comment:
throw new API_Exception('You need to supply a valid Payment ID to delete a payment') rather than create_error 
(we need to get rid of those create_error lines so people stop copying them :-)